src/runtime/types.ts

```diff
@@ -0,0 +1,80 @@
+import type { NitroFetchOptions } from 'nitropack'
+
+export type IgnoreCase<T extends string> = Lowercase<T> | Uppercase<T>
+export type RemovePrefix<T extends string, P extends string> = T extends `${P}${infer S}` ? S : never
+
+export type PathItemObject = { [M in HttpMethod]?: any } & { parameters?: any }
+
+// Constant types
+export type HttpMethod = 'get' | 'put' | 'post' | 'delete' | 'options' | 'head' | 'patch' | 'trace'
+export type OkStatus = 200 | 201 | 202 | 203 | 204 | 206 | 207 | '2XX' | 'default'
+export type ErrorStatus = 500 | '5XX' | 400 | 401 | 402 | 403 | 404 | 405 | 406 | 407 | 408 | 409 | 410 | 411 | 412 | 413 | 414 | 415 | 416 | 417 | 418 | 420 | 421 | 422 | 423 | 424 | 425 | 426 | 429 | 431 | 444 | 450 | 451 | 497 | 498 | 499 | '4XX'
+
+// General purpose types
+export type RequestBody<T> = T extends { requestBody: { content: infer Body } }
+  ?
+    | (Body extends { 'application/octet-stream': any }
+      ? {
+          body: string | Blob
+          headers: { 'content-type': 'application/octet-stream' }
+        }
+      : {})
```

Comment:
Quick question: What do you mean by `{}`? 

We shouldn't use `{}` as a type. `{}` actually means "any non-nullish value", according to ESLint.
- If you want a type meaning "any object", you probably want `object` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.
- If you want a type meaning "empty object", you probably want `Record<string, never>` instead.
...

See the [eslint@typescript-eslint/ban-types](https://typescript-eslint.io/rules/ban-types/).